pkgs/by-name/rp/rpm-sequoia/package.nix

```diff
@@ -0,0 +1,87 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+  darwin,
+  nettle,
+  nix-update-script,
+  rustPlatform,
+  pkg-config,
+  runCommand,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rpm-sequoia";
+  version = "1.7.0";
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = "rpm-sequoia";
+    rev = "v${version}";
+    hash = "sha256-AZCsboUv4muKOw5El2Hw5O1cvAgD3JhBppacrQCJT2k=";
+  };
+
+  cargoHash = "sha256-0yO1+OAkXje/ir8i8URVhIcW8gwXleYx+XL1U4bjtXk=";
+
+  nativeBuildInputs =
+    [
+      pkg-config
+      rustPlatform.bindgenHook
+    ]
+    ++ lib.optionals
+      (stdenv.hostPlatform != stdenv.buildPlatform && stdenv.buildPlatform.canExecute stdenv.hostPlatform)
+      [
+        # Tests will try to shell out to objdump to check the exported symbols
+        # fixup pkgsLLVM.rpm-sequoia
+        (runCommand "objdump" { } ''
+          mkdir -p $out/bin
+          ln -s ${stdenv.cc.bintools}/bin/${stdenv.cc.bintools.targetPrefix}objdump $out/bin/objdump
+        '')
+      ];
+
+  propagatedBuildInputs = [ nettle ];
```

Comment:
Is propagation actually required here?

Reply:
I think so. 

the `.pc` looks like:
```
cat result-dev/lib/pkgconfig/rpm-sequoia.pc
prefix=/nix/store/195zhqd45snw3hyjbibmyw72f8d3c7aj-rpm-sequoia-1.7.0
libdir=${prefix}/lib

Name: rpm-sequoia
Description: An implementation of the RPM PGP interface using Sequoia.
URL: https://sequoia-pgp.org/
Version: 1.7.0
Requires.private: nettle
Cflags:
Libs: -L${libdir} -lrpm_sequoia
```

The consumer (`rpm`) about not finding nettle otherwise.

Reply:
Ah, yeah, our handling of `Requires.private` is weird right now.